packages/datx/src/helpers/model/fields.ts

```diff
@@ -234,7 +238,7 @@ export function getBackRef(model: PureModel, key: string): PureModel | Array<Pur
   }
 
   const backModels = collection
-    .getAllModels()
+    .findAll(getModelRefType(refOptions.model, model, model, key, collection))
```

Comment:
What's the reasoning behind this change? I don't think this can work, since v2 has polymorphic relationship and therefore we can't assume the type of a ref.

Reply:
my initial thought was that every access `getBackRef` in a full collection scan, which might lead to performance degradation. The indirect references to the model are determined at definition time, so initial filtering at scan time should improve performance.

Reply:
If don't filter by Type, then using ID as a basis for judgment is ineffective because ID is not unique (ID & Type is unique)

Reply:
It's entirely possible that the old logic was also wrong - the comparison should be done based on both the id and type. For performance reasons, it might be worth to do it in an array map and findOne since findOne complexity is O(1) and findAll complexity is O(n)

Reply:
I agree with you, but the current situation is that the backref attribute does not store a value. So findOne can't be used directly because you don't know which models are referenced by the backref.
`collection.findAll(Type). filter(Boolean)` is O(n) reasonable solution I can find.

Reply:
How should the current situation be handled?

Reply:
Good point about `findOne`.
Since `hasBackRef` has been updated, I think that this part can remain the same as before (get all and then filter with `hasBackRef`).
I'm not sure it can be optimised further, at least it's not that trivial (without building some indexes for it).

Reply:
I'm sorry I'm not quite sure what you mean. Do you mean that you still use 'collection. GetAllModels ()', and then use 'hasBackRef' to perform '( ID&Type)' filtering? , move 'getModelRefType()' to the 'hasBackRef' method to implement polymorphic type matching?

Reply:
const backModels = collection
    .getAllModels()
    .filter((item) =>
      getModelRefType(refOptions.model, item, model, refOptions.property as string, collection),
    )
    .filter((item) => hasBackRef(item, refOptions.property as string, model));

is right?

Reply:
Why do we need a type check at all? Isn't the `hasBackRef` enough? Especially since we don't know all supported type for a ref - it can be any valid type.

Reply:
My original plan was to have indirect references support ID/IDs, so I directly thought of adding ID to the hasBackRef to discuss the value. In fact, this solution is wrong, because of my cognitive limitations.